raster/src/main/scala/geotrellis/raster/ForEachCell.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2016 Azavea.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import geotrellis.raster._
+import geotrellis.raster.rasterize.{Rasterizer, Callback}
+import geotrellis.vector.Geometry
+
+
+/**
+ * For Geometry Objects
+  */
+trait GeometryForEachCellMethods[T <: Geometry] extends MethodExtensions[T] {
+  import geotrellis.raster.rasterize.{Transformer,Rasterizer}
+
+  def foreachCell(re : RasterExtent, includeExterior : Boolean = false)(fn : Int => Int) : Tile =
+    re.foreachCell(self, includeExterior)(fn)
+
+  def foreachCellDouble(re : RasterExtent, includeExterior : Boolean = false)(fn : Double => Double) : Tile =
+    re.foreachCellDouble(self, includeExterior)(fn)
+
+  def rasterize(re : RasterExtent)(fn : Transformer[Int]) = Rasterizer.rasterize(self, re)(fn)
+
+  def foreachCellCovering(re : RasterExtent)(fn : Int => Int) : Tile = foreachCell(re, true)(fn)
+  def foreachCellContanedBy(re : RasterExtent)(fn : Int => Int) : Tile = foreachCell(re, false)(fn)
+  def foreachCellCoveringDouble(re : RasterExtent)(fn : Double => Double) : Tile = foreachCellDouble(re, true)(fn)
+  def foreachCellContanedByDouble(re : RasterExtent)(fn : Double => Double) : Tile = foreachCellDouble(re, false)(fn)
+}
+
+/**
+ * For Raster Extents
+ */
+trait RasterExtentForEachCellMethods[T <: RasterExtent] extends MethodExtensions[T] {
+  def foreachCell(geom : Geometry, includeExterior : Boolean = false)(fn : Int => Int) : Tile = {
+    val nd = NODATA
+    val mutableTile = IntArrayTile.fill(nd, self.cols, self.rows)
+
+    Rasterizer.foreachCellByGeometry(geom, self, includeExterior)( new Callback {
+      def apply(col : Int, row : Int): Unit = mutableTile.set(col, row, fn(nd))
```

Comment:
Okay, it wasn't clear to me what the intent was of these changes.  I still do not have the full context, but what I have done now is make the Geometry and RasterExtent versions of these methods take a function which takes a column and row so that a tile can be created de novo, whereas the Tile and Raster versions take a value (since they already have associated tiles).